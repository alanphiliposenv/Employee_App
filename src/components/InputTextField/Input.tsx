import { ChangeEvent, FC } from 'react';
import './styles.css';

type InputPropTypes = {
  type: 'text' | 'password';
  placeholder: string;
  label: string;
  value: string;
  onChange: (e: ChangeEvent<HTMLInputElement>) => void;
};

const Input: FC<InputPropTypes> = ({ type, placeholder, label, value, onChange }) => {
  return (
    <div className='text-input'>
      <label>{label}</label>
      <input
        className='.input'
        type={type}
        placeholder={placeholder}
        value={value}
        onChange={onChange}
      />
    </div>
  );
};

export default Input;
