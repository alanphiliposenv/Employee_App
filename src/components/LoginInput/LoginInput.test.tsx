import '@testing-library/jest-dom';
import { fireEvent, render, screen } from '@testing-library/react';
import Input, { InputPropTypes } from './LoginInput';

describe('Tests for LoginInput', () => {
  test('if type correctly rendered as text', () => {
    const InputProps: InputPropTypes = {
      type: 'text'
    };

    render(<Input {...InputProps} />);
    const element = screen.getByTestId('login-input-field');

    expect(element).toHaveAttribute('type', 'text');
    expect(element).not.toHaveAttribute('type', 'password');
  });
  test('if type correctly rendered as password', () => {
    const InputProps: InputPropTypes = {
      type: 'password'
    };

    render(<Input {...InputProps} />);
    const element = screen.getByTestId('login-input-field');

    expect(element).toHaveAttribute('type', 'password');
    expect(element).not.toHaveAttribute('type', 'text');
  });
  test('if label rendered correctly', () => {
    const InputProps: InputPropTypes = {
      label: 'label-text'
    };

    render(<Input {...InputProps} />);
    const element = screen.getByTestId('login-input-label');

    expect(element).toHaveTextContent('label-text');
    expect(element).not.toHaveTextContent('wrong-label');
  });
  test('if value rendered correctly', () => {
    const InputProps: InputPropTypes = {
      value: 'value-text'
    };

    render(<Input {...InputProps} />);
    const element = screen.getByTestId('login-input-field');

    expect(element).toHaveAttribute('value', 'value-text');
    expect(element).not.toHaveAttribute('value', 'wrong-value');
  });
  test('if onChangne rendered correctly', () => {
    const onChange = jest.fn();
    const InputProps: InputPropTypes = {
      value: 'value-text',
      onChange
    };

    render(<Input {...InputProps} />);
    const element = screen.getByTestId('login-input-field');

    expect(element).toHaveAttribute('value', 'value-text');
    fireEvent.change(element, { target: { value: 'new-value-text' } });
    expect(onChange).toHaveBeenCalledTimes(1);
    expect(onChange).toHaveBeenCalledWith('new-value-text');
  });
  test('snapshot redered', () => {
    const InputProps: InputPropTypes = {
      value: 'value-text',
      label: 'label-text'
    };

    const element = render(<Input {...InputProps} />);

    expect(element).toMatchSnapshot();
  });
});
