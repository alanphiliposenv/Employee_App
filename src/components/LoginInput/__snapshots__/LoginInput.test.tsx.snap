// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests for LoginInput snapshot redered 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="login-input"
      >
        <input
          class="login-input-text-field"
          data-testid="login-input-field"
          placeholder=""
          value="value-text"
        />
        <label
          class="login-input-label"
          data-testid="login-input-label"
        >
          label-text
        </label>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="login-input"
    >
      <input
        class="login-input-text-field"
        data-testid="login-input-field"
        placeholder=""
        value="value-text"
      />
      <label
        class="login-input-label"
        data-testid="login-input-label"
      >
        label-text
      </label>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
